conf/gulp-tasks/template/templatetaskfactory.js

```diff
@@ -14,67 +14,87 @@ class TemplateTaskFactory {
   constructor (translationResolver, locale = 'en') {
```

Comment:
I could have sworn I left this comment in the initial review, but not I can't find it anywhere. 

As it stands, this class doesn't follow the archetype of a Factory. The Factory design pattern is a bit prescriptive. Specifically, it exposes a single method to the consumer. This method takes in a type or some other enum and returns an object corresponding to that type. 

I think we need to update this class to fully follow the Factory pattern or name it differently. If we wanted a factory, you could create a TemplateTask enum or something like that. There'd be a method that takes in that enum and spits out the right gulp task for you.

Reply:
refactored into 3 smaller factories! I think the idea of 1 factory doing one kind of thing makes a lot of sense